src/Microsoft.AspNet.OData.Shared/Builder/OperationConfiguration.cs

```diff
@@ -91,7 +93,18 @@ public string Namespace
         /// <summary>
         /// The type returned when the operation is invoked.
         /// </summary>
-        public IEdmTypeConfiguration ReturnType { get; set; }
+        public IEdmTypeConfiguration ReturnType
+        {
+            get { return _returnType; }
+            set
+            {
+                _returnType = value;
+                if (value != null)
+                {
+                    DerivedTypeConstraints.ClrBaseType = value.ClrType;
```

Comment:
>DerivedTypeConstraints.ClrBaseType = value.ClrType; [](start = 20, length = 51)

That's weird and change/add the configuration even no derived type configuration happens in customer scenarios

Reply:
The option was to lazy initialize and save extra allocations or initialize with the constructor. Model builder not being the most performance critical piece, I decided to initialize with the constructor. Since the return type does not have to be specified in the operation configuration initialization,I need to update it for the derivedTypeConstraint whenever it gets set. 

---
In reply to: [394512858](https://github.com/OData/WebApi/pull/1977#discussion_r394512858) [](ancestors = 394512858)